llmfoundry/models/mpt/modeling_mpt.py

```diff
@@ -705,7 +707,25 @@ def fsdp_wrap_fn(self, module: nn.Module) -> bool:
 
     # Activation Checkpointing
     def activation_checkpointing_fn(self, module: nn.Module) -> bool:
-        return isinstance(module, MPTBlock)
+        if not hasattr(self.config, 'activation_checkpointing_target'):
+            return isinstance(module, MPTBlock)
+        act_ckpt_str = self.config.activation_checkpointing_target
+        act_ckpt_lst = act_ckpt_str.replace(' ', '').split(',')
```

Comment:
this list conversion isn't needed. omconfig allows for lists eg adding this in the yaml:
```
activation_checkpointing_target:
  - mptblock
  - grouped_query_attention
```

should result in: 
```config.activation_checkpointing_target = ['mptblock', 'grouped_query_attention']```